### PR TITLE
feat: ✨ roll inititive override

### PR DESCRIPTION
**Pull request by DrOgres**
## Summary
Working on building out the overrides for the Combat class. 

## Checklist
- [x] Roll initiative override
- [X] Sort Combatants
- [X] Draw a card
- [X] Pick a Card -- unsure if this is actually needed
- [X] Find a Card
- [X] Reset all
- [X] start combat
- [X] play initiative sound -- we should get a cool card shuffle sound I think 🎴

### CODE REVIEW

**Comment by Stefouch** [comment]
> Pick a Card -- unsure if this is actually needed

I think it is important: Some YZE games allows to draw more than 1 initiative card, and choose the best (or the worst)

**Comment by DrOgres** [comment]
ok I see ... we still need a way to flag this behavior at some level as I
think the default behavior should be as simple as possible to use from a UX
perspective.
i.e. since most actors draw once that should be the default where an auto
draw will take care of that or a click of the draw button just draws a
single card. and only actors with the appropriate state will draw multiple
and keep, the user is only ever prompted to draw and keep when  it is
mechanically necessary...

File: src/combat/combat.js [hunk 1]
```diff
@@ -11,13 +12,82 @@ export default class YearZeroCombat extends Combat {
...
+      </div>
+      <h4 class="result-text result-text-card">${card?.name}</h4>
+      </section>`;
+
+      const messageData = mergeObject(
+        {
+          speaker: {
+            scene: game.scenes?.active?.id,
+            actor: combatant.actor ? combatant.actor.id : null,
+            alias: `${combatant.token.name} ${game.i18n.localize('YZE.Combat.Initiative.Draw')}`,
```
Comment (Stefouch) [inline src/combat/combat.js:57]:
```suggestion
            alias: game.i18n.format('YZE.Combat.Initiative.Draw', { name: combatant.token.name }),
```
Due to incompatibilities between languages, it's better to put the token name in the translation variable using the `.format` method.

I'll also add the modification for the YML translation files.

File: src/lang/en.yml [hunk 2]
```diff
@@ -3,3 +3,5 @@ YZEC.InitiativeDeckDiscardPileName: Initiative Cards Discard Pile
 YZEC.NoInitiativeDeckFound: No initiative deck found, creating a new one...
 YZEC.NoInitiativeDeckDiscardPileFound: No discard pile found, creating a new one...
 YZEC.WARNING.InvalidDeckType: You can only draw initiative cards from a deck
+YZE.Combat.Initiative.Draw: '''s Initiative Card!'
```
Comment (Stefouch) [inline src/lang/en.yml:6]:
```suggestion
YZE.Combat.Initiative.Draw: '{name}''s Initiative Card!'
```

File: src/lang/fr.yml [hunk 3]
```diff
@@ -3,3 +3,5 @@ YZEC.InitiativeDeckDiscardPileName: Défausse des Cartes d'Initiative
 YZEC.NoInitiativeDeckFound: Aucun deck d'initiative trouvé, création d'un nouveau ...
 YZEC.NoInitiativeDeckDiscardPileFound: Aucune pile de défausse trouvée, création d'une nouvelle ...
 YZEC.WARNING.InvalidDeckType: Vous ne pouvez piocher des cartes d'initiative que depuis un deck
+YZE.Combat.Initiative.NotEnoughCards: Pas assez de cartes dans le paquet pour piocher l'initiative
+YZE.Combat.Initiative.Draw: La carte d'initiative de !
```
Comment (Stefouch) [inline src/lang/fr.yml:7]:
```suggestion
YZE.Combat.Initiative.Draw: La carte d'initiative de {name} !
```

**Review by Stefouch — APPROVED** [review]

**Pull request merged**